internal/util/projutil/exec.go

```diff
@@ -48,11 +48,11 @@ type GoCmdOptions struct {
 	Env []string
 	// Dir is the dir to run "go {cmd}" in; exec.Command.Dir is set to this value.
 	Dir string
-	// GoMod determines whether to set the "-mod=vendor" flag.
-	// If true, "go {cmd}" will use modules.
-	// If false, "go {cmd}" will not use go modules. This is the default.
+	// NoGoMod determines whether to set the "-mod=vendor" flag.
+	// If true, "go {cmd}" will not use modules.
+	// If false, "go {cmd}" will use go modules. This is the default.
 	// This applies to build, clean, get, install, list, run, and test.
-	GoMod bool
+	NoGoMod bool
```

Comment:
@estroz Does this to be set based on `projutil.GetDepManagerType()` when we call `projutil.GoBuild()` and `projutil.GoTest()` (i.e. if the dep manager type is "dep", do we need to set `NoGoMod=true`)? For example, when `operator-sdk up local` builds the go binary:

https://github.com/operator-framework/operator-sdk/blob/3c8e5e7280d8b8584e46605faba999e4fa3c4487/cmd/operator-sdk/up/local.go#L184-L189